ext/srt/gstsrtobject.c

```diff
@@ -829,6 +851,48 @@ thread_func (gpointer data)
   }
 }
 
+static GSocketAddress *
+peeraddr_to_g_socket_address (const struct sockaddr *peeraddr) {
+  gsize peeraddr_len;
+
+  switch (peeraddr->sa_family) {
+    case AF_INET:
+      peeraddr_len = sizeof (struct sockaddr_in);
+      break;
+    case AF_INET6:
+      peeraddr_len = sizeof (struct sockaddr_in6);
+      break;
+    default:
+      g_warning ("Unsupported address family %d", peeraddr->sa_family);
+      return NULL;
+  }
+  return g_socket_address_new_from_native ((gpointer) peeraddr, peeraddr_len);
+}
+
+static gint
+srt_listen_callback_func(GstSRTObject * self, SRTSOCKET sock, int hs_version,
+    const struct sockaddr* peeraddr, const char* stream_id) {
+  g_autoptr (GSocketAddress) addr = peeraddr_to_g_socket_address (peeraddr);
```

Comment:
I think at least MSVC doesn't support `g_autoptr` and the like. We couldn't care less for MSVC in Hwangsaeul, but in upstream GStreamer autocleanup is better avoided.

Reply:
Yes, the Hwangsaeul project supports only the Linux platform and the latest GLib, but GStreamer-wise, we should follow their rules.